tests/test_sensor.py

```diff
@@ -262,3 +307,72 @@ def test_remove_callbacks(self, ss, sensors, add_callback, remove_callback):
         ss.remove(sensors[0])
         remove_callback.assert_not_called()
         add_callback.assert_not_called()
+
+
+class MyAgg(AggregateSensor):
+    """A simple AggregateSensor subclass for testing."""
+
+    def update_aggregate(
+        self,
+        updated_sensor: Optional[Sensor],
+        reading: Optional[Reading],
+        old_reading: Optional[Reading],
+    ) -> Reading:
+        """Return a known Reading."""
+        return Reading(0, Sensor.Status.NOMINAL, 7)
+
+
+@pytest.fixture
+def agg_sensor(mocker, ss):
+    """Mock out update_aggregate so we can check it's called appropriately."""
+
+    def return_a_reading(*args, **kwargs):
+        """Return an arbitrary Reading."""
+        return Reading(0, Sensor.Status.WARN, 8)
+
+    my_agg = MyAgg(target=ss, sensor_type=int, name="good-bad-ugly")
+    mocker.patch.object(my_agg, "update_aggregate", side_effect=return_a_reading)
```

Comment:
I'm wondering if it would work just say `side_effect=my_agg.update_aggregate` and remove `return_a_reading`, so that you don't have to define two arbitrary readings, and it continues to call the original function.

Reply:
After a quick test it does work. I kind of like my approach though because then you can see that the reading is actually changed from the original in response to some kind of change in the input. If they're just the same, then you can't distinguish.

Unless I put some logic in `MyAgg.update_aggregate()` to check for the various `None`s. Which eliminates `return_a_reading()` but one would still need to define a few arbitrary readings.